alloy/src/main/java/com/liferay/faces/alloy/taghandler/LoadConstants.java

```diff
@@ -90,37 +91,70 @@ public LoadConstants(TagConfig config) throws Exception {
 	}
 
 	@Override
-	public void apply(FaceletContext facesContext, UIComponent parentUIComponent) throws IOException {
+	public void apply(FaceletContext facletContext, UIComponent parentUIComponent) throws IOException {
+
+		Cache<String, Map<String, Object>> constantCache = LoadConstants.constantCache;
+
+		// First check without locking (not yet thread-safe)
+		if (constantCache == null) {
+
+			// Second check with locking (thread-safe)
+			synchronized (LoadConstants.class) {
```

Comment:
@ngriffin7a, they are created when you hit the view that they are on. 1 is created per `alloy:loadConstant` on the page.